### DISCUSSION TITLE
Usar o firebase para simular um backend em produção

### DISCUSSION THREAD

**Discussion by jlucaso1** [prpontes/lp4_appusuarios#2] [Ideas]
Estava pensando na possibilidade de migrar o projeto para o firebase e descontinuar o sqflite ( já que na maioria dos projetos não é recomendado para produção por poder ser decompilado ). Com o firebase teriamos acesso ao firebase-auth (autenticação com tokens) e o firestore (banco de dados no-sql de alta perfomance e tempo real).

### Vantagens:

- Desacomplamento de regra de negócio no lado do cliente.
- Segurança ao guardar dados sensiveis.
- Sincronização de todos os dados ao decorrer do projeto.

### Desvantagens:

- Precisará de aulas para entender o funcionamento e como utilizar o firebase no flutter.
- Precisará de acesso a internet para funcionar.

**Comment by prpontes** [comment]
Sim, João Lucas! A ideia é que nós possamos migrar ao poucos o projeto para a plataforma Firebase. Para isso, nas aulas seguintes da nossa disciplina, estaremos avançando rumo a esse objetivo. Inicialmente o objetivo foi apresentar banco de dados sqlite para persistência de dados, próximo passo será o consumo de API e posteriormente entraremos no assunto do Firebase.
